README.md

```diff
@@ -295,6 +295,140 @@ dpkg --list | grep chefdk # or dpkg --status chefdk
+
+### Test Versions
+
+chef-dk is tested by the [chef-acceptance framework](https://github.com/chef/chef-acceptance), which contains suites that are run on the Jenkins test machines. The definitions of the tests are in the `acceptance` directory. The version of chef-acceptance and test-kitchen, are governed by `acceptance/Gemfile`.
+
+The test tooling versions are locked down with `acceptance/Gemfile.lock`, which can be updated by running `rake dependencies`.
+
+## The Build Process
+
+The actual ChefDK build process is done with Omnibus, and has several general steps:
+
+1. `bundle install` from `chef-dk/Gemfile.lock`
+2. Reinstall any gems that came from git or path using `rake install`
+3. appbundle chef, chef-dk, test-kitchen and berkshelf
+4. Put miscellaneous powershell scripts and cleanup
+
+### Kicking Off The Build
+
+The build is kicked off in Jenkins by running this on the machine (which is already the correct OS and already has the correct dependencies, loaded by the `omnibus` cookbook):
+
+```
+load-omnibus-toolchain.bat
+cd chef-dk/omnibus
+bundle install
+bundle exec omnibus build chefdk
+```
+
+This causes the [chefdk project definition](omnibus/config/projects/chefdk.rb) to load, which runs the [chef-dk-complete](omnibus/config/software/chef-dk-complete.rb) software definition, the primary software definition driving the whole build process. The reason we embed it all in a software definiton instead of the project is to take advantage of omnibus caching: omnibus will invalidate the entire project (and recompile ruby, openssl, and everything else) if you change anything at all in the project file. Not so with a software definition.
+
+### Installing the Gems
+
+The primary build definition that installs the many ChefDK rubygems is [`software/chef-dk.rb`](omnibus/software/chef-dk.rb). This has dependencies on any binary libraries, ruby, rubygems and bundler. It has a lot of steps, so it uses a [library](omnibus/files/chef-dk/build-chef-dk.rb) to help reuse code and make it manageable to look at.
+
+What it does:
+
+1. Depends on software defs for pre-cached gems (see "Gems and Caching" below).
+2. Installs all gems from the bundle:
+   - Sets up a `.bundle/config` ([code](omnibus/files/chef-dk/build-chef-dk.rb#L17-L39)) with --retries=4, --jobs=1, --without=development,no_<platform>, and `build.config.nokogiri` to pass.
+   - Sets up a common environment, standardizing the compilers and flags we use, in [`env`](omnibus/files/chef-dk-gem/build-chef-dk-gem.rb#L32-L54).
+   - [Runs](omnibus/config/software/chef-dk.rb#L68) `bundle install --verbose`
+3. Reinstalls any gems that were installed via path:
+   - [Runs](omnibus/files/chef-dk/build-chef-dk.rb#L80) `bundle list --paths` to get the installed directories of all gems.
+   - For each gem not installed in the main gem dir, [runs](omnibus/files/chef-dk/build-chef-dk.rb#L89) `rake install` from the installed gem directory.
```

Comment:
So if we install `test-kitchen` from a custom branch it will to end up in `.../gems/bundler/test-kitchen-123456`.  This step ensures that `test-kitchen` gets installed to the main `gems` folder and doesn't have a weird directory name, correct?